### ISSUE TITLE
More than one version installed (per backend)?

### ISSUE THREAD

**Issue by fibbo** [emscripten-core/emsdk#477]
Is it not possible anymore to keep more than one version of a backend? Let's say I have 1.39.12 (upstream) installed and run `emsdk install 1.39.13`. The old version is then gone. Do I have to manage versions myself?

**Comment by kripken** [comment]
We removed this after users reported problems with disk space - each version takes hundreds of MB, so if we keep all the versions forever eventually that becomes a problem.

It would be good to fix this if you or someone else has time. One option is to keep versions but issue a warning when the disk space gets high, with instructions for how to clear it. Another option might be to save the last N versions but no more.

**Comment by sbc100** [comment]
I'd like to see this behaviour restored if possible.

**Comment by flab03** [comment]
So, when installing a new version, the old version is silently erased, but the situation is actually worse than this. Every version from 1.38.33 onwards is marked INSTALLED. Running `emsdk activate <old_version>` doesn't fail. So in effect emsdk is lying to me when it says that it activated an old version when it didn't. This is extremely counterintuitive and I lost a lot of time figuring this out, especially since README.md says

> You can toggle between different tools and SDK versions by running emsdk activate <tool/sdk name>. 

If the new policy is to keep only one version, then at least make sure that activating a version other than the installed version fails.

**Comment by kripken** [comment]
Sorry about that @flab03 , thanks for the bug report! I opened #505 which I think will fix the reporting.

I also verified that with that patch, trying to activate a non-installed tool will skip it and warn:
```
Warning: The SDK/tool 'releases-upstream-40f3caabcef7b52bdde63d3883462414d7a25bec-64bit' cannot be activated since it is not installed! Skipping this tool...
```

**Comment by torarnv** [comment]
I think this is still an issue. I was under the impression, based on the distinction between `install` and `activate`, that the emsdk repo/tool could be used to mange multiple parallel installations, but in practice this does not seem to be the case.

The tool however happily allows me to install `tot-upstream` and `3.1.25` and activate them both, without actually changing anything in `upstream/emscripten`. 

And `list` shows both versions installed.

**Comment by sbc100** [comment]
Its is still the case that you can only install one pre-build sdk at a time.  Each installation will replace the previous one that lives in the `upstream` directory.

I'd like find a way to do parallel installations of SDK versions (like you can for different versions of node, python and other tools in emsdk).   But currently this is not possible.

**Comment by MoshiKoi** [comment]
I found this issue after searching for a bit of how to install multiple versions, since the README implies it is possible.

> So, when installing a new version, the old version is silently erased, but the situation is actually worse than this. Every version from 1.38.33 onwards is marked INSTALLED. Running `emsdk activate <old_version>` doesn't fail. So in effect emsdk is lying to me when it says that it activated an old version when it didn't. This is extremely counterintuitive and I lost a lot of time figuring this out, especially since README.md says
> 
> > You can toggle between different tools and SDK versions by running emsdk activate <tool/sdk name>.
> 
> If the new policy is to keep only one version, then at least make sure that activating a version other than the installed version fails.

I think the README should probably be updated if this is no longer a supported use case. There is no way to "toggle" between different tools with `activate` - given they all get installed to upstream, in fact, just installing the other version is sufficient since it still points to the same place.

**Comment by sbc100** [comment]
Yes, its a little confusing, because parallel installations of all of the other Tools and SDKs *is* supported.  Its only the prebuilt sdk itself that is no longer supported with parallel installation.
